src/sortStudents.ts

```diff
@@ -1,16 +1,62 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
-export enum SortType {
-  // describe SortType enum
+export enum SortType{
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc'|'desc';
 
+function getAverageGrade(grades: number[]):number {
+  const sum:number = grades.reduce((a, b) => a + b, 0);
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  return sum / grades.length;
+}
+
+export function sortStudents(
+  students: Student[], sortBy: SortType, order: SortOrder,
+) :Student[] {
+  const copyStudents:Student[] = [...students];
+
+  if (sortBy === SortType.Name || sortBy === SortType.Surname) {
+    if (order === 'asc') {
+      return copyStudents.sort((a:Student,
+        b:Student) => a[sortBy].localeCompare(b[sortBy]));
+    }
+
+    return copyStudents.sort((a:Student,
+      b:Student) => b[sortBy].localeCompare(a[sortBy]));
```

Comment:
consider to use ternary
   here and in other cases
   
```suggestion
      return copyStudents.sort((a:Student,
        b:Student) => order === 'asc' 
          ? a[sortBy].localeCompare(b[sortBy]))
           : b[sortBy].localeCompare(a[sortBy]));
```